File: swift/problem754.swift
//simple math problem
class Solution {
    func reachNumber(_ target: Int) -> Int {
        var target = abs(target)
        var k = 0
        
        while (target > 0) {
            k += 1
            target -= k
        }
        
        return target&1 == 0 ? k : (k+1+(k&1))
    }
}

File: python/problem754.py
# from solution
class Solution:
    def reachNumber(self, target: int) -> int:
        target = abs(target)
        sumn = 0
        n = 0
        while sumn < target:
            n += 1
            sumn += n
        
        k = sumn - target
        if k%2 == 0 :
            return n  
        elif (k + n + 1) % 2 ==0:
                return n+1
        else:
            return n+2
        